tests/python/pants_test/backend/jvm/subsystems/test_jvm_platform.py

```diff
@@ -200,3 +200,79 @@ def test_synthetic_target_runtime_platform_lookup(self):
+                        "platform-without-jvm-options": {"target": "8"},
+                    },
+                }
+            },
+        )
+        instance = JvmPlatform.global_instance()
+        with_options = instance.get_platform_by_name("platform-with-jvm-options")
+        without_options = instance.get_platform_by_name("platform-without-jvm-options")
+
+        assert ("-Dsomething",) == with_options.jvm_options
+        assert tuple() == without_options.jvm_options
+
+    def test_jvm_options_from_platform_shlexed(self):
+        init_subsystem(
+            JvmPlatform,
+            options={
+                "jvm-platform": {
+                    "platforms": {
+                        "platform-with-shlexable-vm-options": {
+                            "target": "8",
+                            "jvm_options": ["-Dsomething -Dsomethingelse"],
+                        },
+                    },
+                }
+            },
+        )
+        instance = JvmPlatform.global_instance()
+        need_shlex_options = instance.get_platform_by_name("platform-with-shlexable-vm-options")
+
+        assert ("-Dsomething", "-Dsomethingelse") == need_shlex_options.jvm_options
+
+    def test_compile_setting_equivalence(self):
+        assert JvmPlatformSettings(
+            source_level="11", target_level="11", args=["-Xfoo:bar"], jvm_options=[]
+        ) == JvmPlatformSettings(
+            source_level="11", target_level="11", args=["-Xfoo:bar"], jvm_options=[]
+        )
+        assert JvmPlatformSettings(
+            source_level="11", target_level="11", args=[], jvm_options=["-Xfoo:bar"]
+        ) == JvmPlatformSettings(
+            source_level="11", target_level="11", args=[], jvm_options=["-Xfoo:bar"]
+        )
+
+    def test_compile_setting_inequivalence(self):
+        assert JvmPlatformSettings(
+            source_level="11", target_level="11", args=[], jvm_options=[]
+        ) != JvmPlatformSettings(source_level="11", target_level="12", args=[], jvm_options=[])
+
+        assert JvmPlatformSettings(
+            source_level="11", target_level="11", args=["-Xfoo:bar"], jvm_options=[]
+        ) != JvmPlatformSettings(
+            source_level="11", target_level="11", args=["-XSomethingElse"], jvm_options=[]
+        )
+
+        assert JvmPlatformSettings(
+            source_level="9", target_level="11", args=[], jvm_options=[]
+        ) != JvmPlatformSettings(source_level="11", target_level="11", args=[], jvm_options=[])
+
+        assert JvmPlatformSettings(
+            source_level="11", target_level="11", args=[], jvm_options=["-Xvmsomething"]
+        ) != JvmPlatformSettings(source_level="11", target_level="11", args=[], jvm_options=[])
+
+    def test_incompatible_source_target_level_raises_error(self):
+        with self.assertRaises(JvmPlatformSettings.IllegalSourceTargetCombination):
+            JvmPlatformSettings(source_level="11", target_level="9", args=[], jvm_options=[])
```

Comment:
nit: is it intentional to use
`assert .. == ..`
instead of 
`self.assertEqual...`?

Reply:
I was asked to use `assert ==` because our test suite now supports displaying failures from them in a more fluent way.

Reply:
ah thanks. good to know!